src/main/java/co/cask/hydrator/plugin/DateTransform.java

```diff
@@ -134,9 +135,13 @@ public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter)
             return;
           }
         } else {
-          throw new IllegalArgumentException(String.format("Cannot parse value %s for format %s. %s.",
-                                                           input.get(sourceField), config.targetFormat, e.getMessage()),
-                                             e);
+          if (ignoreErrors.toLowerCase().equals("yes")) {
+            builder.set(targetField, input.get(sourceField));
```

Comment:
This only works if the source and target fields are of the same type. If the source is a long and target is a string, this will break.

Also, is this really the desired behavior? It doesn't seem helpful, as you won't be able to tell which records in the output have been correctly formatted and which have not. I think it would make more sense for the option to either emit these records as an error, or to set the value to null.

By the way, the existing code above is really indirect and messy way of handling null input values. Nulls should be handled in the original loop starting at line 94, not here when catching a generic Exception and checking various conditions to guess that the Exception is a NullPointerException.